### PR TITLE
Add shortcuts to agent keys option.

### PR DESCRIPTION
**Pull request by benyissa**
image

### CODE REVIEW

**Review by ErebusZ — DISMISSED** [review]

File: src/ostorlab/cli/scan/run/run.py [hunk 1]
```diff
@@ -32,8 +32,10 @@
 @click.option(
     "--agent",
     multiple=True,
-    help="List of agents keys. to use in the scan.",
+    help="To include an agent in the scan, add its key. The expected formats are: key (ostorlab organization will be "
+    "used by default) | @org/key | agent/org/key",
```
Comment (3asm) [inline src/ostorlab/cli/scan/run/run.py:36]:
```suggestion
    help="List agents keys (agent/<org>/<name> or @<org>/<name>). Org name can be omitted for defaults agent hosted by Ostorlab.",
```

File: src/ostorlab/cli/types.py [hunk 2]
```diff
@@ -49,3 +51,50 @@ def convert(
...
+        """
+        try:
+            forward_slash_count = cli_value.count("/")
+            if forward_slash_count == 0:
+                return f"{OSTORLAB_AGENTS_PREFIX}/{cli_value}"
+            elif forward_slash_count == 1:
+                if cli_value.startswith("@"):
+                    return f"agent/{cli_value[1:]}"
+                else:
+                    raise ValueError
```
Comment (3asm) [inline src/ostorlab/cli/types.py:89]:
```suggestion
                    raise ValueError("Invalid agent key value.")
```

File: src/ostorlab/cli/types.py [hunk 3]
```diff
@@ -49,3 +51,50 @@ def convert(
...
+                return cli_value
+            else:
+                raise ValueError
+        except ValueError:
+            self.fail(
+                message=f"Invalid agent key: {cli_value}. The expected formats are: key (ostorlab organization will "
+                f"be used by default) | @org/key | agent/org/key",
+                param=param,
+                ctx=ctx,
+            )
```
Comment (3asm) [inline src/ostorlab/cli/types.py:100]: Raise to fail. What kind of pattern is that :open_mouth: .

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/cli/scan/run/run.py [hunk 4]
```diff
@@ -32,8 +32,10 @@
 @click.option(
     "--agent",
     multiple=True,
-    help="List of agents keys. to use in the scan.",
+    help="List agents keys (agent/<org>/<name> or @<org>/<name>). Org name can be omitted for defaults agent hosted "
+    "by Ostorlab.",
     required=False,
+    type=types.AgentKeyTpe(),
```
Comment (deadly-panda) [inline src/ostorlab/cli/scan/run/run.py:38]: You mean AgentKeyT**y**pe. Please fix!

File: tests/cli/scan/run/assets/ip_test.py [hunk 5]
```diff
@@ -151,3 +153,48 @@ def testScanRunFile_whenPassedArgIsOfTypeArrayAndSupportedByTheAgent_callScanWit
...
+
+@pytest.mark.parametrize(
+    "cli_agent_key, expected_agent_key",
+    [
+        ("nmap", "agent/ostorlab/nmap"),
+        ("@dev/nmap", "agent/dev/nmap"),
+        ("agent/test/nmap", "agent/test/nmap"),
+    ],
+)
+def testScanRunFile_whenAgentKeyPassed_callScanWithValidSettings(
```
Comment (deadly-panda) [inline tests/cli/scan/run/assets/ip_test.py:166]: update the test name, this is not scanning a file.

File: tests/cli/scan/run/run_test.py [hunk 6]
```diff
@@ -383,3 +383,28 @@ def testOstorlabScanRunCLI_whenWrongArgsFormatProvided_showsErrorMessage() -> No
         "Invalid argument test,test. The expected format is name:value."
         in result.output
     )
+
+
+@pytest.mark.parametrize(
+    "invalid_agent_key",
+    ["/nmap", "@agent/ostorlab/nmap/"],
+)
```
Comment (deadly-panda) [inline tests/cli/scan/run/run_test.py:391]: Are you adding more cases? If not, why `parametrize`.

**Review by deadly-panda — CHANGES_REQUESTED** [review]

**Review by elyousfi5 — APPROVED** [review]
lgtm

**Review by 3asm — APPROVED** [review]

**Pull request merged**